File: testers/avro/scala-cats/generated-and-checked-in/com/example/service/UserService.scala
package com.example.service

import cats.effect.IO
import java.lang.Void

/** User management service protocol */
trait UserService {
  /** Get a user by their ID */
  def getUser(userId: String): IO[Result[User, UserNotFoundError]]

  /** Create a new user */
  def createUser(
    email: String,
    name: String
  ): IO[Result[User, ValidationError]]

  /** Delete a user */
  def deleteUser(userId: String): IO[Result[Unit, UserNotFoundError]]

  /** Send a notification to a user (fire-and-forget) */
  def notifyUser(
    userId: String,
    message: String
  ): IO[Void]
}

File: testers/avro/java-spring/generated-and-checked-in/com/example/service/UserService.java
package com.example.service;

/** User management service protocol */
public interface UserService {
  /** Get a user by their ID */
  Result<User, UserNotFoundError> getUser(String userId);

  /** Create a new user */
  Result<User, ValidationError> createUser(String email, String name);

  /** Delete a user */
  Result<Void, UserNotFoundError> deleteUser(String userId);

  /** Send a notification to a user (fire-and-forget) */
  void notifyUser(String userId, String message);
}
